### PR TITLE
Related refactor

### PR DESCRIPTION
**Pull request by bishalkg**
Refactored related components to use App component props passed in. Also refactored user list card to include localStorage for persistence of selected cards.

### CODE REVIEW

File: client/src/components/RelatedItemsAndOutfit/ListCard.jsx [hunk 1]
```diff
@@ -1,43 +1,61 @@
...
       <div className="card-details-container">
         <h6 className="category-heading">
           {prodInfo.category}
         </h6>
-        <p className="product-name-p">{prodInfo.name}</p>
+        <p className="product-name-p">
+          {prodInfo.name}
+          <br />
+          <span className="style-name-span">
+            {`(${styleName})`}
```
Comment (ya-liu) [inline client/src/components/RelatedItemsAndOutfit/ListCard.jsx:74]: {styleName} suffices?

File: client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx [hunk 2]
```diff
@@ -7,35 +8,36 @@ class RelatedItemsAndOutfit extends React.Component {
...
-    // console.log(productId);
-    this.setState({
-      currProdId: productId,
-    });
+  setCurrProdToState() {
+    const {
+      currProdId, prodInfo, prodStyles, prodReviewsMeta,
+    } = this.props;
+    this.setState({ currProd: [prodInfo, prodStyles, prodReviewsMeta] });
+    this.getRelatedData(currProdId);
```
Comment (ya-liu) [inline client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx]: I recommend putting this line as a callback to this.setState, setState takes an optional CB to fire immediately after the states are set. https://stackoverflow.com/questions/42038590/when-to-use-react-setstate-callback

File: client/src/components/RelatedItemsAndOutfit/UserOutfits/UserOutfit.jsx [hunk 3]
```diff
@@ -1,26 +1,46 @@
...
 
-const UserOutfit = ({ currProd, onRelatedCardClick }) => {
+const UserOutfit = ({ currProd, changeProductHandler, prodStyleSelected }) => {
   const [currOutfits, setCurrOutfits] = useState([]);
+  const [styleIds, setStyleIds] = useState([]);
+
+  useEffect(() => {
+    if (!currOutfits.length) {
+      const currStorage = [];
+      Object.entries(localStorage).forEach((entry) => currStorage.push(JSON.parse(entry[1])));
```
Comment (ya-liu) [inline client/src/components/RelatedItemsAndOutfit/UserOutfits/UserOutfit.jsx:14]: I had so much trouble with modifying a state array in useEffect yesterday. This is very clear and well-written.

**Review by dho1994 — APPROVED** [review]

**Review by ya-liu — APPROVED** [review]

File: client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx (hunk 2 above)
Comment (bishalkg) [reply client/src/components/RelatedItemsAndOutfit/RelatedItemsAndOutfit.jsx]: done! thanks

**Pull request merged**